packages/quantic/force-app/main/default/lwc/quanticSearchBoxInput/__tests__/quanticSearchBoxInput.test.js

```diff
@@ -0,0 +1,344 @@
+      const element = createTestComponent({
+        ...defaultOptions,
+        withoutSubmitButton: false,
+      });
+      await flushPromises();
+
+      const searchIcon = element.shadowRoot.querySelector(
+        selectors.searchBoxSearchIcon
+      );
+
+      expect(searchIcon).toBeNull();
+    });
+
+    it('should display the submit button to the right of the searchbox', async () => {
+      const element = createTestComponent({
+        ...defaultOptions,
+        withoutSubmitButton: false,
+      });
+      await flushPromises();
+
+      const submitButton = element.shadowRoot.querySelector(
+        selectors.searchBoxSubmitBtn
+      );
+
+      expect(submitButton).not.toBeNull();
+    });
+  });
+
+  describe('when the withoutSubmitButton is set to true', () => {
+    it('should display the searchIcon to the left of the searchbox', async () => {
+      const element = createTestComponent({
+        ...defaultOptions,
+        withoutSubmitButton: true,
+      });
+      await flushPromises();
+
+      const searchIcon = element.shadowRoot.querySelector(
+        selectors.searchBoxSearchIcon
+      );
+
+      expect(searchIcon).not.toBeNull();
+      expect(searchIcon.classList.contains('slds-input__icon_left')).toBe(true);
+    });
+
+    it('should not display the submit button to the right of the searchbox', async () => {
+      const element = createTestComponent({
+        ...defaultOptions,
+        withoutSubmitButton: true,
+      });
+      await flushPromises();
+
+      const submitButton = element.shadowRoot.querySelector(
+        selectors.searchBoxSubmitBtn
+      );
+
+      expect(submitButton).toBeNull();
+    });
+  });
+
+  describe('when the placeholder property receives a custom placeholder value', () => {
+    it('should display the custom value as the searchbox placeholder', async () => {
+      const customPlaceholder = 'Custom placeholder';
+      const element = createTestComponent({
+        ...defaultOptions,
+        placeholder: customPlaceholder,
+      });
+      await flushPromises();
+
+      const input = element.shadowRoot.querySelector(selectors.searchBoxInput);
+
+      expect(input.placeholder).not.toEqual(defaultOptions.placeholder);
+      expect(input.placeholder).toEqual(customPlaceholder);
+    });
+  });
+
+  describe('when suggestions are found', () => {
+    it('should display the suggestions in the suggestions list', async () => {
+      const element = createTestComponent({
+        ...defaultOptions,
+        suggestions: mockSuggestions,
+      });
+      await flushPromises();
+
+      const suggestionsList = element.shadowRoot.querySelector(
+        selectors.searchBoxSuggestionsList
+      );
+      expect(suggestionsList).not.toBeNull();
+
+      const suggestionsListItems =
+        suggestionsList.shadowRoot.querySelectorAll('li');
+      expect(suggestionsListItems).not.toBeNull();
+
+      expect(suggestionsListItems.length).toEqual(3);
```

Comment:
Where is that 3 coming from? Would be better to have it as the `.length` of the array of mockSuggestions